File: Transport-Web-Angular/src/app/services/camion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Camion } from '../core/models';

/**
 * NB: le backend actuel n'expose PAS encore /api/camions (voir AUDIT.md).
 * Ce service est prêt — il fonctionnera dès l'ajout du CamionController.
 */
@Injectable({ providedIn: 'root' })
export class CamionService {
  private base = `${environment.apiUrl}/camions`;
  constructor(private http: HttpClient) {}

  getAll(): Observable<Camion[]> { return this.http.get<Camion[]>(this.base); }
  getById(id: number): Observable<Camion> { return this.http.get<Camion>(`${this.base}/${id}`); }
  create(c: Partial<Camion>): Observable<Camion> { return this.http.post<Camion>(this.base, c); }
  update(id: number, c: Partial<Camion>): Observable<Camion> { return this.http.put<Camion>(`${this.base}/${id}`, c); }
  delete(id: number): Observable<void> { return this.http.delete<void>(`${this.base}/${id}`); }
}

File: Transport-BackEnd/src/main/java/com/agileo/transport/service/CamionService.java
package com.agileo.transport.service;

import com.agileo.transport.Dtos.request.CamionRequestDTO;
import com.agileo.transport.Dtos.response.CamionResponseDTO;

import java.util.List;

public interface CamionService {
    List<CamionResponseDTO> getAll();
    CamionResponseDTO getById(Long id);
    CamionResponseDTO create(CamionRequestDTO dto);
    CamionResponseDTO update(Long id, CamionRequestDTO dto);
    void delete(Long id);
}
